src/test/java/com/nvidia/spark/rapids/jni/DecimalUtilsTest.java

```diff
@@ -86,6 +86,22 @@ void simplePosMultiplyZeroByNegOne() {
     }
   }
 
+  @Test
+  void multiply128WithoutInterimCast() {
+    try (ColumnVector lhs =
+                 makeDec128Column("-8533444864753048107770677711.1312637916");
+         ColumnVector rhs =
+                 makeDec128Column("-12.0000000000");
+         ColumnVector expectedBasic =
+                 makeDec128Column("102401338377036577293248132533.575165");
+         ColumnVector expectedValid =
+                 ColumnVector.fromBooleans(false);
+         Table found = DecimalUtils.multiply128(lhs, rhs, -6, false)) {
```

Comment:
Should there be an explicit test for the interim cast as well?

Reply:
Yeah, there are. All the multiply128 tests call into the one with `interimCast=true`. Here is one example https://github.com/NVIDIA/spark-rapids-jni/blob/f6bb2052af9d3bce60d087c758c76b6d77b9936f/src/test/java/com/nvidia/spark/rapids/jni/DecimalUtilsTest.java#L74